### ISSUE TITLE
Custom hairs on races are overridden through faction hairTags

### ISSUE THREAD

**Issue by Solid-Wires** [Solid-Wires/rimworldmod-pawnkindracediversification#8]
5katz claims that some aliens that are supposed to have specified hair or no hair still spawn on pawnkinds that were originally human.
Now I actually have not experienced this problem at all through my tests. This could be a problem from another mod... But, if I were to have witnessed this issue, then I would have listed it as an unfixable bug.

![image](https://user-images.githubusercontent.com/72309175/115242217-a7eeb900-a0ef-11eb-93e3-d457e837467c.png)
Now how hair works for races in vanilla is that their custom faction def specifies what hairtags their pawnkinds use. This entails that hairs are specified through factions alone in vanilla. If this was still the case today, then this issue would not have been fixable unless _HAR would allow modders to specify hairs somehow..._

![image](https://user-images.githubusercontent.com/72309175/115242535-f603bc80-a0ef-11eb-93e0-bc962ac778db.png)
...And apparently they can! Although this is a bit of a weird example (coming from the [Bun Race](https://steamcommunity.com/sharedfiles/filedetails/?id=2108324996)), any race can specify what hair their race's pawns spawn with through the hair settings in hairTags.

![image](https://user-images.githubusercontent.com/72309175/115243243-b9849080-a0f0-11eb-9e85-bec24c12c16f.png)
Even [Ratkin](https://steamcommunity.com/sharedfiles/filedetails/?id=1578693166) does this with their hairs as well. Therefore, it doesn't matter what faction any alien race spawns in because their hair tags are specified under their hair settings, which override the hair tags of the faction.

![image](https://user-images.githubusercontent.com/72309175/115243527-049ea380-a0f1-11eb-9c2c-eac414580f31.png)
[Android tiers](https://steamcommunity.com/sharedfiles/filedetails/?id=1386412863), which have androids who are not supposed to have hair, will not generate hair because the <hasHair /> boolean was set to false, even if they are in a faction that has hair tags. This example is from a T1 android.

The results:
![image](https://user-images.githubusercontent.com/72309175/115244605-12085d80-a0f2-11eb-9374-27fe7f4f5ffa.png)
The left with backfacing pawns are tribal warriors. The right are colonists. They are spawning with proper hairs on my end.

The only thing that I can see that would be causing that problem would be that the race author did not properly add hair tags to their race. Races like the [Orassans](https://steamcommunity.com/sharedfiles/filedetails/?id=1541519487) typically use vanilla urban hairstyles in their own faction, but since they did not have race-specific hair (nothing specified in hairTags), any faction that they generate in will assign them the faction-specific hairs (urban hairs, tribal hairs, etc). Therefore, this does not appear to be an issue associated with this mod, and if this is happening, then it is working as intended from this explanation.

**Comment by Solid-Wires** [comment]
Another thing to note specifically about the Bun Race is that they have races _and_ hairs specifically tied to tribals. This is very strange... If you were to specify tribal Buns in global weights, you would therefore see this "race" everywhere. If the mod author wants to specify that this race only appears in a tribal faction, then they would have to patch their own race weights extension.

**Comment by Solid-Wires** [comment]
I discovered something that I think 5katz was talking about. If you look closer at this image:
![image](https://user-images.githubusercontent.com/72309175/115244605-12085d80-a0f2-11eb-9374-27fe7f4f5ffa.png)
It appears that some races are using hairs that these factions specified. Okay, now this is certainly a bug that I can fix. It appears that the hairs are _not overridden, but added_ towards the pool of hairs that the pawns will generate with. Yeah, I can fix that.

LATER EDIT: This assumption is incorrect. HAR completely replaces the hair tags that the pawn's generating faction has. The hairs are not added to a pool. Race-specific hair is truly race-specific.

**Comment by Solid-Wires** [comment]
So with using this race, I discovered a few things:
  Alien_BunTribe does NOT specify hair tags (therefore, they do not spawn with any race-specific hair in particular)
      -These are the aliens that you are seeing in the image that uses tribal hairs.
  Alien_BunTribeNew are the aliens that use specific hair defs. For some reason, the author left the former old race in there.
  Alien_Bunlander are aliens similar to the new Tribal Buns.

It appears that this is an unexpected issue with some alien races that do this sort of behavior. I tried hitting into the hair generator, but HAR already sets the specifics on hairs for us already - during pawn generation. This does not appear like an issue I can fix on my side. My former theory seems to be correct in that this is an issue with the alien race, and not an issue with how I handle generation.
In this specific instance, if the author of the Bun race does not intend for the old alien race to show up anywhere, then they would have to patch an extension to the race and set the flat weight to -1 to prevent it from being adjusted anywhere.
